packages/dev/core/test/unit/DeviceInput/babylon.deviceInput.test.ts

```diff
@@ -0,0 +1,226 @@
+import { DeviceSource, DeviceSourceManager, DeviceType, PointerInput } from "core/DeviceInput";
+import type { IDeviceInputSystem } from "core/DeviceInput/InputDevices/inputInterfaces";
+import type { InternalDeviceSourceManager, IObservableManager } from "core/DeviceInput/InputDevices/internalDeviceSourceManager";
+import { NullEngine } from "core/Engines/nullEngine";
+import type { IPointerEvent, IUIEvent, IWheelEvent } from "core/Events";
+import type { Nullable } from "core/types";
+
+/**
+ * This class mocks up the basic functionality required from the DeviceInputSystem
+ */
+class MockDeviceInputSystem implements IDeviceInputSystem {
+    constructor(
+        public onDeviceConnected: Nullable<(deviceType: DeviceType, deviceSlot: number) => void>,
+        public onDeviceDisconnected: Nullable<(deviceType: DeviceType, deviceSlot: number) => void>,
+        public onInputChanged: Nullable<(deviceType: DeviceType, deviceSlot: number, eventData: IUIEvent) => void>
+    ) {}
+    pollInput(deviceType: DeviceType, deviceSlot: number, inputIndex: number): number {
+        if (deviceType === DeviceType.Mouse && deviceSlot === 0 && inputIndex === PointerInput.LeftClick) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+    isDeviceAvailable(deviceType: DeviceType): boolean {
+        return deviceType === DeviceType.Mouse || deviceType === DeviceType.Touch;
+    }
+    dispose(): void {
+        // Do nothing
+    }
+}

```

Comment:
What is the goal of this class in terms of functionality? Right now, it looks like the `pollInput` is being mocked to always be in a state where left mouse button is down? I'm assuming this will evolve as you add more tests?

Reply:
Also, are you unable to use the [mock functionality from Jest](https://jestjs.io/docs/es6-class-mocks)?